### ISSUE TITLE
Rebooted upstreams hide downstreams' data until they come back online

### ISSUE THREAD

**Issue by kousu** [netdata/netdata#10627]
##### Bug report summary

If a [streaming parent](https://learn.netdata.cloud/docs/agent/streaming#database-replication) restarts, it temporarily loses track of its children until they reconnect, denying me access to review the state of the children.

Crashing children are, for me, the most important data; I want to be able to review their archives to be able to diagnose them. But netdata shows me the opposite set in this case.

##### OS / Environment

```
Linux monitor.neuro.polymtl.ca 5.4.0-65-generic #73-Ubuntu SMP Mon Jan 18 17:25:17 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=20.04
/etc/lsb-release:DISTRIB_CODENAME=focal
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 20.04.2 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="20.04.2 LTS (Focal Fossa)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 20.04.2 LTS"
/etc/os-release:VERSION_ID="20.04"
/etc/os-release:HOME_URL="https://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="https://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
/etc/os-release:PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
/etc/os-release:VERSION_CODENAME=focal
/etc/os-release:UBUNTU_CODENAME=focal
```

##### Netdata version

 
netdata v1.19.0

##### Component Name

web? streaming? database? all three?

##### Steps To Reproduce

I followed https://learn.netdata.cloud/docs/agent/streaming#database-replication to set up a parent/child relationship, intending the parent to help me debug the child when the child crashes. I also disabled `delete obsolete charts files` and `delete orphan hosts files` (per https://github.com/netdata/netdata/issues/7360#issuecomment-557796791).

These are my configs:

```
root@monitor:~# cat /etc/netdata/netdata.conf

[global]
	run as user = netdata
	web files owner = root
	web files group = root
	bind socket to IP = 0.0.0.0

	# according to https://learn.netdata.cloud/docs/store/change-metrics-storage#calculate-the-system-resources-ram-disk-space-needed-to-store-metrics
	# this should store data for about 2 years for 2 nodes
	update every = 15
	dbengine disk space = 50000

	delete obsolete charts files = no
	delete orphan hosts files = no

root@monitor:~# cat /etc/netdata/stream.conf 
[1a89d18f-e7b5-4847-957d-9a45e8a4467f]
    enabled = yes
    default memory mode = dbengine
    allow from = 192.168.99.*
    health enabled by default = auto
    default postpone alarms on connect seconds = 60
    multiple connections = allow
```

```
root@data:~# cat /etc/netdata/netdata.conf 
[global]
	run as user = netdata
	web files owner = root
	web files group = root
	bind socket to IP = 0.0.0.0

        update every = 1
        memory mode = dbengine
        dbengine disk space = 1000

root@data:~# cat /etc/netdata/stream.conf 
[stream]
    enabled = yes
    destination = 192.168.99.2
    api key = 1a89d18f-e7b5-4847-957d-9a45e8a4467f
    timeout seconds = 60
    default port = 19999
    send charts matching = *
    buffer size bytes = 1048576
    reconnect delay seconds = 5
    initial clock resync iterations = 60
```

It works great and I have this menu:

![2021-02-14-044333_495x311_scrot](https://user-images.githubusercontent.com/987487/107873553-6a687800-6e81-11eb-9928-96d85fd29775.png)

If the children crash, I can still inspect their logs, and if they come back, they pick up where they left off, just leaving a gap in the data:

```
root@data:~# systemctl stop netdata
root@data:~# sleep 5
root@data:~# systemctl start netdata
```

![2021-02-14-044250_1068x526_scrot](https://user-images.githubusercontent.com/987487/107873566-7bb18480-6e81-11eb-95a4-0cc940d0e043.png)

However, if the parent crashes during the child's outage:

```
root@data:~# systemctl stop netdata
```

```
root@monitor:~# systemctl restart netdata
```

then there is no way to get back to the child:

1. The option is gone from the menu

    ![2021-02-14-044448_494x198_scrot](https://user-images.githubusercontent.com/987487/107873597-aac7f600-6e81-11eb-826c-781a1bef7f3e.png)

2. Trying the URL `http://<parent>:19999/host/<child>/` directly gets

    > This netdata does not maintain a database for host: <child>

    ![2021-02-14-044458_1026x133_scrot](https://user-images.githubusercontent.com/987487/107873604-b2879a80-6e81-11eb-850e-a57db4e2da72.png)

The forgetting happens even if the child nodes are explicitly listed in the parent's `/etc/netdata/stream.conf` like

```
# unhelpful attempt to force netdata to look into it's offline database
[675df3e8-6e80-11eb-8287-4b05b8b3ad74]
    enabled = yes
```

**Keep in mind**, while the child is down, the parent **does** know about it, at least, it knows about this:

```
root@monitor:~# head -n 1 /var/lib/netdata/675df3e8-6e80-11eb-8287-4b05b8b3ad74/health/health-log.db
U	data.neuro.polymtl.ca	6028d095	6028d8ad	00000002	6028d0dd	6028d069	6028d8cf	0000003c	00000000	90000003	00000000	6028d8cf	10s_ipv4_tcp_resets_received	ipv4.tcphandshake	tcp		sysadmin	56@/usr/lib/netdata/conf.d/health.d/tcp_resets.conf	tcp resets/s	average TCP RESETS this host is receiving, over the last 10 seconds (this can be an indication that a service this host needs, has crashed; clear notification for this alarm will not be sent)	0	10	0	0.000000	nan	0000000000000000
```

and it knows the old data, since it's still there once the child does come back:

![2021-02-14-044603_1086x617_scrot](https://user-images.githubusercontent.com/987487/107873711-5709dc80-6e82-11eb-855e-89c34482bfc3.png)

so the "does not maintain a database" line is misleading.

Thank you for your software, and for your time picking through all my screenshots.

##### Expected behavior

Could all the archived hosts and their data be found and displayed,whether or not it's just-booted or it has been online for awhile, and even if the children haven't been updating in a while? After all, netdata is perfectly content to show archived data about a host that is temporarily offline; it's only inconsistent if it itself also goes offline temporarily.

Eventually, their data can age out, and then they can be deleted, but until they hit my retention limits I want to be able to see them at all times!

**Comment by stelfrag** [comment]
> Crashing children are, for me, the most important data; I want to be able to review their archives to be able to diagnose them. But netdata only offers me the opposite set in this case. I just have to hope really hard that my monitoring servers don't crash too, and I don't want to rely on that.

Hi @kousu 

I apologize for the delayed reply. Work already started in order to support this functionality so stayed tuned as we are working out the details.

**Comment by kousu** [comment]
No worries. I know how it goes. I have lots of my own projects on the go and most of them fall by the wayside.

This isn't a crisis for us, it's just something we hope to have fixed eventually to have better coverage. I am happy to know this is in the queue now.

Thanks for all your work.

**Comment by kousu** [comment]
> Could all the archived hosts and their data be found and displayed,whether or not it's just-booted or it has been online for awhile, and even if the children haven't been updating in a while? After all, netdata is perfectly content to show archived data about a host that is temporarily offline; it's only inconsistent if it itself also goes offline temporarily.

and also after all, netdata seems to *keep* the archived data just fine, because once a child reconnects the archive reappears. The data never starts from where the child booted.

**Comment by kousu** [comment]
Actually I need to revise that last statement about netdata holding a persistent archive. Tonight I had a server hang and while hung it didn't show up in menu picker, but I could go to my bookmark for it and see its archived data up to 8:22, but after rebooting everything before 10:02 was erased. Here's proof:

Before:

![Screenshot from 2021-04-09 21-49-00](https://user-images.githubusercontent.com/987487/114255810-ba485480-9983-11eb-9c32-a857fd4c1f1c.png)

After:

![Screenshot from 2021-04-09 22-11-25](https://user-images.githubusercontent.com/987487/114255811-bae0eb00-9983-11eb-86da-09fe4da7c9b5.png)

But I'm running `netdata` as packaged in Ubuntu-LTS (`1.19.0-3ubuntu1`) which is over a year old so maybe this has already been fixed.

If I can reproduce it in a lab setting I guess I'll submit a bug report about it, but I probably won't have the time for that in the short term.

**Comment by cakrit** [comment]
@stelfrag this seems to be linked to the ability to display charts that aren't live, so it's either a duplicate or can be linked to a PR in progress that will resolve it.

**Comment by kousu** [comment]
> @stelfrag this seems to be linked to the ability to display charts that aren't live, so it's either a duplicate or can be linked to a PR in progress that will resolve it.

Yes, that seems about right! I want to be able to see unlive charts, machines that haven't connected. Though eventually I want machines that I've fully taken out of service, or machines that were ephemeral, to get deleted. I have trust that those PRs in progress have a way to handle tuning what counts as permanent vs ephemeral.
